gcr.go

```diff
@@ -0,0 +1,121 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"os/exec"
+	"path"
+	"sort"
+	"strings"
+)
+
+type gcrAuthResponse struct {
+	Token     string `json:"token"`
+	ExpiresIn int    `json:"expires_in"`
+}
+
+type gcrTagsResponse struct {
+	Manifest map[string]json.RawMessage `json:"manifest"`
+}
+
+// for sorting setups
+type gcrImageDetail struct {
+	TimeCreatedMs string   `json:"timeCreatedMs"`
+	Tag           []string `json:"tag"`
+}
+
+type gcrImages []gcrImageDetail
+
+// ref: https://stackoverflow.com/questions/34037256/does-google-container-registry-support-docker-remote-api-v2/34046435#34046435
+func fetchBearer(repo string, image string) (string, error) {
+	token, err := exec.Command("gcloud", "auth", "print-access-token").Output() // run gcloud command
+	if err != nil {
+		return "", err
+	}
+
+	url := constructGCRAuthURL(repo, image)
+	body, err := httpGet(url, "", &BasicAuthInfo{
+		Username: "_token",
+		Password: strings.TrimSpace(string(token)), // need to remove trailing new line character
```

Comment:
I would rather not use the external command to reduce dependencies. You can use [`FindDefaultCredentials`](https://pkg.go.dev/golang.org/x/oauth2/google?tab=doc#FindDefaultCredentials) instead.

```suggestion
	credential, err := google.FindDefaultCredentials(context.Background())
	if err != nil {
		return "", err
	}

	token, err := credential.TokenSource.Token()
	if err != nil {
		return "", err
	}

	url := constructGCRAuthURL(repo, image)
	body, err := httpGet(url, "", &BasicAuthInfo{
		Username: "_token",
		Password: token.AccessToken,
```

Reply:
Thanks! 
I agree with this.
> I would rather not use the external command to reduce dependencies

fixed in 872f215